backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -929,23 +945,204 @@ func addDisplayNameColumn(db *gorm.DB, mdl interface{}, dialect SQLDialect) erro
+		if err := ensureMinioBucketExists(ctx, config, bucketName, blobConfig.accessKey, blobConfig.secretKey, k8sClient); err != nil {
+			glog.Warningf("Failed to ensure MinIO bucket exists (may already exist): %v", err)
+		}
+	}
+
+	glog.Infof("Successfully initialized blob storage for bucket: %s", bucketName)
+	return storage.NewBlobObjectStore(bucket, pipelinePath), nil
+}
+
+// shouldEnsureObjectBucket determines whether we should proactively ensure the
+// object bucket exists. We do this for S3-compatible providers (e.g., MinIO)
+// when a bucket name is configured and SessionInfo is available.
+func shouldEnsureObjectBucket(bucketName string, config *objectstore.Config) bool {
+	if bucketName == "" || config == nil || config.SessionInfo == nil {
+		return false
+	}
+	provider := strings.ToLower(strings.TrimSpace(config.SessionInfo.Provider))
+	return provider == "minio" || provider == "s3"
+}
+
+// blobStorageConfig holds both the objectstore config and credentials
+type blobStorageConfig struct {
+	config          *objectstore.Config
+	accessKey       string
+	secretKey       string
+	useDirectBucket bool // indicates if direct blob.OpenBucket should be used (MinIO with env credentials)
+}
+
+// ensureProtocol adds http:// or https:// protocol if not present
+func ensureProtocol(endpoint string, secure bool) string {
+	if strings.HasPrefix(endpoint, "http://") || strings.HasPrefix(endpoint, "https://") {
+		return endpoint
+	}
+	protocol := "http://"
+	if secure {
+		protocol = "https://"
+	}
+	return protocol + endpoint
+}
+
+// buildConfigFromEnvVars creates objectstore.Config from environment variables
+// This bridges the gap between API server's environment variable configuration
+// and the v2 objectstore package's expected configuration format.
+// Returns both the config and credentials to avoid re-reading them.
+func buildConfigFromEnvVars() *blobStorageConfig {
+	bucketName := common.GetStringConfigWithDefault("ObjectStoreConfig.BucketName", "")
+	host := common.GetStringConfigWithDefault("ObjectStoreConfig.Host", "")
+	port := common.GetStringConfigWithDefault("ObjectStoreConfig.Port", "")
+	secure := common.GetBoolConfigWithDefault("ObjectStoreConfig.Secure", false)
+	region := common.GetStringConfigWithDefault("ObjectStoreConfig.Region", "")
+	accessKey := common.GetStringConfigWithDefault("ObjectStoreConfig.AccessKey", "")
+	secretKey := common.GetStringConfigWithDefault("ObjectStoreConfig.SecretAccessKey", "")
+
+	// Constants for MinIO secret (consistent with v2 config)
+	const minioArtifactSecretName = "mlpipeline-minio-artifact"
+	const minioArtifactAccessKeyKey = "accesskey"
+	const minioArtifactSecretKeyKey = "secretkey"
+
+	// Set AWS environment variables
+	if accessKey != "" {
+		os.Setenv("AWS_ACCESS_KEY_ID", accessKey)
+	}
+	if secretKey != "" {
+		os.Setenv("AWS_SECRET_ACCESS_KEY", secretKey)
+	}
+	// Set region - use default for MinIO if not specified
+	isMinIO := host != ""
```

Comment:
minio is no longer the default and the host needs to always be provided, in fact let's run validation on these fields: BucketName / Host / AccessKey / SecretAccessKey, these are always required.

Reply:
Done.